File: qt/gen_qabstractslider.go
package qt

import (
	"unsafe"
)

type QAbstractSlider__SliderAction int

const (
	QAbstractSlider__SliderNoAction      QAbstractSlider__SliderAction = 0
	QAbstractSlider__SliderSingleStepAdd QAbstractSlider__SliderAction = 1
	QAbstractSlider__SliderSingleStepSub QAbstractSlider__SliderAction = 2
	QAbstractSlider__SliderPageStepAdd   QAbstractSlider__SliderAction = 3
	QAbstractSlider__SliderPageStepSub   QAbstractSlider__SliderAction = 4
	QAbstractSlider__SliderToMinimum     QAbstractSlider__SliderAction = 5
	QAbstractSlider__SliderToMaximum     QAbstractSlider__SliderAction = 6
	QAbstractSlider__SliderMove          QAbstractSlider__SliderAction = 7
)

type QAbstractSlider__SliderChange int

const (
	QAbstractSlider__SliderRangeChange       QAbstractSlider__SliderChange = 0
	QAbstractSlider__SliderOrientationChange QAbstractSlider__SliderChange = 1
	QAbstractSlider__SliderStepsChange       QAbstractSlider__SliderChange = 2
	QAbstractSlider__SliderValueChange       QAbstractSlider__SliderChange = 3
)

type QAbstractSlider struct {
	h          uintptr
	isSubclass bool
}

// NewQAbstractSlider constructs a new QAbstractSlider object.
func NewQAbstractSlider(parent *QWidget) *QAbstractSlider {
	ret := newQAbstractSlider(QAbstractSlider_new(parent.cPointer()))
	ret.isSubclass = true
	return ret
}

// NewQAbstractSlider2 constructs a new QAbstractSlider object.
func NewQAbstractSlider2() *QAbstractSlider {
	ret := newQAbstractSlider(QAbstractSlider_new2())
	ret.isSubclass = true
	return ret
}

func (this *QAbstractSlider) MetaObject() *QMetaObject {
	return newQMetaObject(QAbstractSlider_MetaObject(this.h))
}

func (this *QAbstractSlider) Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))
	return (unsafe.Pointer)(QAbstractSlider_Metacast(this.h, param1_Cstring))
}

func QAbstractSlider_Tr(s string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	var _ms struct_miqt_string = QAbstractSlider_Tr(s_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QAbstractSlider) Orientation() Orientation {
	return (Orientation)(QAbstractSlider_Orientation(this.h))
}

func (this *QAbstractSlider) SetMinimum(minimum int) {
	QAbstractSlider_SetMinimum(this.h, (int)(minimum))
}

func (this *QAbstractSlider) Minimum() int {
	return (int)(QAbstractSlider_Minimum(this.h))
}

func (this *QAbstractSlider) SetMaximum(maximum int) {
	QAbstractSlider_SetMaximum(this.h, (int)(maximum))
}

func (this *QAbstractSlider) Maximum() int {
	return (int)(QAbstractSlider_Maximum(this.h))
}

func (this *QAbstractSlider) SetSingleStep(singleStep int) {
	QAbstractSlider_SetSingleStep(this.h, (int)(singleStep))
}

func (this *QAbstractSlider) SingleStep() int {
	return (int)(QAbstractSlider_SingleStep(this.h))
}

func (this *QAbstractSlider) SetPageStep(pageStep int) {
	QAbstractSlider_SetPageStep(this.h, (int)(pageStep))
}

func (this *QAbstractSlider) PageStep() int {
	return (int)(QAbstractSlider_PageStep(this.h))
}

func (this *QAbstractSlider) SetTracking(enable bool) {
	QAbstractSlider_SetTracking(this.h, (bool)(enable))
}

func (this *QAbstractSlider) HasTracking() bool {
	return (bool)(QAbstractSlider_HasTracking(this.h))
}

func (this *QAbstractSlider) SetSliderDown(sliderDown bool) {
	QAbstractSlider_SetSliderDown(this.h, (bool)(sliderDown))
}

func (this *QAbstractSlider) IsSliderDown() bool {
	return (bool)(QAbstractSlider_IsSliderDown(this.h))
}

func (this *QAbstractSlider) SetSliderPosition(sliderPosition int) {
	QAbstractSlider_SetSliderPosition(this.h, (int)(sliderPosition))
}

func (this *QAbstractSlider) SliderPosition() int {
	return (int)(QAbstractSlider_SliderPosition(this.h))
}

func (this *QAbstractSlider) SetInvertedAppearance(invertedAppearance bool) {
	QAbstractSlider_SetInvertedAppearance(this.h, (bool)(invertedAppearance))
}

func (this *QAbstractSlider) InvertedAppearance() bool {
	return (bool)(QAbstractSlider_InvertedAppearance(this.h))
}

func (this *QAbstractSlider) SetInvertedControls(invertedControls bool) {
	QAbstractSlider_SetInvertedControls(this.h, (bool)(invertedControls))
}

func (this *QAbstractSlider) InvertedControls() bool {
	return (bool)(QAbstractSlider_InvertedControls(this.h))
}

func (this *QAbstractSlider) Value() int {
	return (int)(QAbstractSlider_Value(this.h))
}

func (this *QAbstractSlider) TriggerAction(action SliderAction) {
	QAbstractSlider_TriggerAction(this.h, action)
}

func (this *QAbstractSlider) SetValue(value int) {
	QAbstractSlider_SetValue(this.h, (int)(value))
}

func (this *QAbstractSlider) SetOrientation(orientation Orientation) {
	QAbstractSlider_SetOrientation(this.h, (int)(orientation))
}

func (this *QAbstractSlider) SetRange(min int, max int) {
	QAbstractSlider_SetRange(this.h, (int)(min), (int)(max))
}

func (this *QAbstractSlider) ValueChanged(value int) {
	QAbstractSlider_ValueChanged(this.h, (int)(value))
}

func (this *QAbstractSlider) OnValueChanged(slot func(value int)) {
	QAbstractSlider_connect_ValueChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QAbstractSlider_ValueChanged
func miqt_exec_callback_QAbstractSlider_ValueChanged(cb intptr_t, value int) {
	gofunc, ok := cgo.Handle(cb).Value().(func(value int))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := (int)(value)

	gofunc(slotval1)
}

func (this *QAbstractSlider) SliderPressed() {
	QAbstractSlider_SliderPressed(this.h)
}

func (this *QAbstractSlider) OnSliderPressed(slot func()) {
	QAbstractSlider_connect_SliderPressed(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QAbstractSlider_SliderPressed
func miqt_exec_callback_QAbstractSlider_SliderPressed(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func (this *QAbstractSlider) SliderMoved(position int) {
	QAbstractSlider_SliderMoved(this.h, (int)(position))
}

func (this *QAbstractSlider) OnSliderMoved(slot func(position int)) {
	QAbstractSlider_connect_SliderMoved(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QAbstractSlider_SliderMoved
func miqt_exec_callback_QAbstractSlider_SliderMoved(cb intptr_t, position int) {
	gofunc, ok := cgo.Handle(cb).Value().(func(position int))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := (int)(position)

	gofunc(slotval1)
}

func (this *QAbstractSlider) SliderReleased() {
	QAbstractSlider_SliderReleased(this.h)
}

func (this *QAbstractSlider) OnSliderReleased(slot func()) {
	QAbstractSlider_connect_SliderReleased(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QAbstractSlider_SliderReleased
func miqt_exec_callback_QAbstractSlider_SliderReleased(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func (this *QAbstractSlider) RangeChanged(min int, max int) {
	QAbstractSlider_RangeChanged(this.h, (int)(min), (int)(max))
}

func (this *QAbstractSlider) OnRangeChanged(slot func(min int, max int)) {
	QAbstractSlider_connect_RangeChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QAbstractSlider_RangeChanged
func miqt_exec_callback_QAbstractSlider_RangeChanged(cb intptr_t, min int, max int) {
	gofunc, ok := cgo.Handle(cb).Value().(func(min int, max int))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := (int)(min)

	slotval2 := (int)(max)

	gofunc(slotval1, slotval2)
}

func (this *QAbstractSlider) ActionTriggered(action int) {
	QAbstractSlider_ActionTriggered(this.h, (int)(action))
}

func (this *QAbstractSlider) OnActionTriggered(slot func(action int)) {
	QAbstractSlider_connect_ActionTriggered(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QAbstractSlider_ActionTriggered
func miqt_exec_callback_QAbstractSlider_ActionTriggered(cb intptr_t, action int) {
	gofunc, ok := cgo.Handle(cb).Value().(func(action int))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := (int)(action)

	gofunc(slotval1)
}

func QAbstractSlider_Tr2(s string, c string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QAbstractSlider_Tr2(s_Cstring, c_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func QAbstractSlider_Tr3(s string, c string, n int) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QAbstractSlider_Tr3(s_Cstring, c_Cstring, (int)(n))
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QAbstractSlider) callVirtualBase_MetaObject() *QMetaObject {
	return newQMetaObject(QAbstractSlider_virtualbase_MetaObject(unsafe.Pointer(this.h)))
}

func (this *QAbstractSlider) OnMetaObject(slot func(super func() *QMetaObject) *QMetaObject) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QAbstractSlider_override_virtual_MetaObject(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QAbstractSlider_MetaObject
func miqt_exec_callback_QAbstractSlider_MetaObject(self QAbstractSlider, cb intptr_t) *QMetaObject {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func() *QMetaObject) *QMetaObject)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	virtualReturn := gofunc((&QAbstractSlider{h: self}).callVirtualBase_MetaObject)

	return virtualReturn.cPointer()
}

func (this *QAbstractSlider) callVirtualBase_Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))

	return (unsafe.Pointer)(QAbstractSlider_virtualbase_Metacast(unsafe.Pointer(this.h), param1_Cstring))
}

func (this *QAbstractSlider) OnMetacast(slot func(super func(param1 string) unsafe.Pointer, param1 string) unsafe.Pointer) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QAbstractSlider_override_virtual_Metacast(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QAbstractSlider_Metacast
func miqt_exec_callback_QAbstractSlider_Metacast(self QAbstractSlider, cb intptr_t, param1 *const_char) unsafe.Pointer {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(param1 string) unsafe.Pointer, param1 string) unsafe.Pointer)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	param1_ret := param1
	slotval1 := GoString(param1_ret)

	virtualReturn := gofunc((&QAbstractSlider{h: self}).callVirtualBase_Metacast, slotval1)

	return virtualReturn
}

File: qt/gen_qabstractslider.cpp
// +build ignore

#include <QAbstractSlider>
#include <QEvent>
#include <QKeyEvent>
#include <QMetaObject>
#include <QObject>
#include <QPaintDevice>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QTimerEvent>
#include <QWheelEvent>
#include <QWidget>
#include <qabstractslider.h>
#include "gen_qabstractslider.h"
class MiqtVirtualQAbstractSlider : public virtual QAbstractSlider {
public:
MiqtVirtualQAbstractSlider(QWidget* parent): QAbstractSlider(parent) {};
MiqtVirtualQAbstractSlider(): QAbstractSlider() {};

virtual ~MiqtVirtualQAbstractSlider() = default;
// cgo.Handle value for overwritten implementation
	intptr_t handle__MetaObject = 0;
// Subclass to allow providing a Go implementation
	virtual const QMetaObject* metaObject() const override {
if (handle__MetaObject == 0) {
return QAbstractSlider::metaObject();
}
QMetaObject* callback_return_value = miqt_exec_callback_QAbstractSlider_MetaObject(const_cast<MiqtVirtualQAbstractSlider*>(this), handle__MetaObject);

		return callback_return_value;
	}
// Wrapper to allow calling protected method
	QMetaObject* virtualbase_MetaObject() const {

		return (QMetaObject*) QAbstractSlider::metaObject();

	}
// cgo.Handle value for overwritten implementation
	intptr_t handle__Metacast = 0;
// Subclass to allow providing a Go implementation
	virtual void* qt_metacast(const char* param1) override {
if (handle__Metacast == 0) {
return QAbstractSlider::qt_metacast(param1);
}
const char* sigval1 = (const char*) param1;

		void* callback_return_value = miqt_exec_callback_QAbstractSlider_Metacast(this, handle__Metacast, sigval1);

		return callback_return_value;
	}
// Wrapper to allow calling protected method
	void* virtualbase_Metacast(const char* param1) {

		return QAbstractSlider::qt_metacast(param1);

	}
};
QAbstractSlider* QAbstractSlider_new(QWidget* parent) {
return new MiqtVirtualQAbstractSlider(parent);
}
QAbstractSlider* QAbstractSlider_new2() {
return new MiqtVirtualQAbstractSlider();
}
void QAbstractSlider_virtbase(QAbstractSlider* src
, QWidget** outptr_QWidget
) {
*outptr_QWidget = static_cast<QWidget*>(src);
}
QMetaObject* QAbstractSlider_MetaObject(const QAbstractSlider* self) {
	return (QMetaObject*) self->metaObject();
}
void* QAbstractSlider_Metacast(QAbstractSlider* self, const char* param1) {
	return self->qt_metacast(param1);
}
struct miqt_string QAbstractSlider_Tr(const char* s) {
	QString _ret = QAbstractSlider::tr(s);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
int QAbstractSlider_Orientation(const QAbstractSlider* self) {
	Qt::Orientation _ret = self->orientation();
	return static_cast<int>(_ret);
}
void QAbstractSlider_SetMinimum(QAbstractSlider* self, int minimum) {
	self->setMinimum(static_cast<int>(minimum));
}
int QAbstractSlider_Minimum(const QAbstractSlider* self) {
	return self->minimum();
}
void QAbstractSlider_SetMaximum(QAbstractSlider* self, int maximum) {
	self->setMaximum(static_cast<int>(maximum));
}
int QAbstractSlider_Maximum(const QAbstractSlider* self) {
	return self->maximum();
}
void QAbstractSlider_SetSingleStep(QAbstractSlider* self, int singleStep) {
	self->setSingleStep(static_cast<int>(singleStep));
}
int QAbstractSlider_SingleStep(const QAbstractSlider* self) {
	return self->singleStep();
}
void QAbstractSlider_SetPageStep(QAbstractSlider* self, int pageStep) {
	self->setPageStep(static_cast<int>(pageStep));
}
int QAbstractSlider_PageStep(const QAbstractSlider* self) {
	return self->pageStep();
}
void QAbstractSlider_SetTracking(QAbstractSlider* self, bool enable) {
	self->setTracking(enable);
}
bool QAbstractSlider_HasTracking(const QAbstractSlider* self) {
	return self->hasTracking();
}
void QAbstractSlider_SetSliderDown(QAbstractSlider* self, bool sliderDown) {
	self->setSliderDown(sliderDown);
}
bool QAbstractSlider_IsSliderDown(const QAbstractSlider* self) {
	return self->isSliderDown();
}
void QAbstractSlider_SetSliderPosition(QAbstractSlider* self, int sliderPosition) {
	self->setSliderPosition(static_cast<int>(sliderPosition));
}
int QAbstractSlider_SliderPosition(const QAbstractSlider* self) {
	return self->sliderPosition();
}
void QAbstractSlider_SetInvertedAppearance(QAbstractSlider* self, bool invertedAppearance) {
	self->setInvertedAppearance(invertedAppearance);
}
bool QAbstractSlider_InvertedAppearance(const QAbstractSlider* self) {
	return self->invertedAppearance();
}
void QAbstractSlider_SetInvertedControls(QAbstractSlider* self, bool invertedControls) {
	self->setInvertedControls(invertedControls);
}
bool QAbstractSlider_InvertedControls(const QAbstractSlider* self) {
	return self->invertedControls();
}
int QAbstractSlider_Value(const QAbstractSlider* self) {
	return self->value();
}
void QAbstractSlider_TriggerAction(QAbstractSlider* self, SliderAction action) {
	self->triggerAction(action);
}
void QAbstractSlider_SetValue(QAbstractSlider* self, int value) {
	self->setValue(static_cast<int>(value));
}
void QAbstractSlider_SetOrientation(QAbstractSlider* self, int orientation) {
	self->setOrientation(static_cast<Qt::Orientation>(orientation));
}
void QAbstractSlider_SetRange(QAbstractSlider* self, int min, int max) {
	self->setRange(static_cast<int>(min), static_cast<int>(max));
}
void QAbstractSlider_ValueChanged(QAbstractSlider* self, int value) {
	self->valueChanged(static_cast<int>(value));
}
void QAbstractSlider_connect_ValueChanged(QAbstractSlider* self, intptr_t slot) {
	MiqtVirtualQAbstractSlider::connect(self, static_cast<void (QAbstractSlider::*)(int)>(&QAbstractSlider::valueChanged), self, [=](int value) {
		int sigval1 = value;
		miqt_exec_callback_QAbstractSlider_ValueChanged(slot, sigval1);
	});
}
void QAbstractSlider_SliderPressed(QAbstractSlider* self) {
	self->sliderPressed();
}
void QAbstractSlider_connect_SliderPressed(QAbstractSlider* self, intptr_t slot) {
	MiqtVirtualQAbstractSlider::connect(self, static_cast<void (QAbstractSlider::*)()>(&QAbstractSlider::sliderPressed), self, [=]() {
		miqt_exec_callback_QAbstractSlider_SliderPressed(slot);
	});
}
void QAbstractSlider_SliderMoved(QAbstractSlider* self, int position) {
	self->sliderMoved(static_cast<int>(position));
}
void QAbstractSlider_connect_SliderMoved(QAbstractSlider* self, intptr_t slot) {
	MiqtVirtualQAbstractSlider::connect(self, static_cast<void (QAbstractSlider::*)(int)>(&QAbstractSlider::sliderMoved), self, [=](int position) {
		int sigval1 = position;
		miqt_exec_callback_QAbstractSlider_SliderMoved(slot, sigval1);
	});
}
void QAbstractSlider_SliderReleased(QAbstractSlider* self) {
	self->sliderReleased();
}
void QAbstractSlider_connect_SliderReleased(QAbstractSlider* self, intptr_t slot) {
	MiqtVirtualQAbstractSlider::connect(self, static_cast<void (QAbstractSlider::*)()>(&QAbstractSlider::sliderReleased), self, [=]() {
		miqt_exec_callback_QAbstractSlider_SliderReleased(slot);
	});
}
void QAbstractSlider_RangeChanged(QAbstractSlider* self, int min, int max) {
	self->rangeChanged(static_cast<int>(min), static_cast<int>(max));
}
void QAbstractSlider_connect_RangeChanged(QAbstractSlider* self, intptr_t slot) {
	MiqtVirtualQAbstractSlider::connect(self, static_cast<void (QAbstractSlider::*)(int, int)>(&QAbstractSlider::rangeChanged), self, [=](int min, int max) {
		int sigval1 = min;
		int sigval2 = max;
		miqt_exec_callback_QAbstractSlider_RangeChanged(slot, sigval1, sigval2);
	});
}
void QAbstractSlider_ActionTriggered(QAbstractSlider* self, int action) {
	self->actionTriggered(static_cast<int>(action));
}
void QAbstractSlider_connect_ActionTriggered(QAbstractSlider* self, intptr_t slot) {
	MiqtVirtualQAbstractSlider::connect(self, static_cast<void (QAbstractSlider::*)(int)>(&QAbstractSlider::actionTriggered), self, [=](int action) {
		int sigval1 = action;
		miqt_exec_callback_QAbstractSlider_ActionTriggered(slot, sigval1);
	});
}
struct miqt_string QAbstractSlider_Tr2(const char* s, const char* c) {
	QString _ret = QAbstractSlider::tr(s, c);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
struct miqt_string QAbstractSlider_Tr3(const char* s, const char* c, int n) {
	QString _ret = QAbstractSlider::tr(s, c, static_cast<int>(n));
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
void QAbstractSlider_override_virtual_MetaObject(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQAbstractSlider*>( (QAbstractSlider*)(self) )->handle__MetaObject = slot;
}
QMetaObject* QAbstractSlider_virtualbase_MetaObject(const void* self) {
	return ( (const MiqtVirtualQAbstractSlider*)(self) )->virtualbase_MetaObject();
}
void QAbstractSlider_override_virtual_Metacast(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQAbstractSlider*>( (QAbstractSlider*)(self) )->handle__Metacast = slot;
}
void* QAbstractSlider_virtualbase_Metacast(void* self, const char* param1) {
	return ( (MiqtVirtualQAbstractSlider*)(self) )->virtualbase_Metacast(param1);
}
void QAbstractSlider_Delete(QAbstractSlider* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<MiqtVirtualQAbstractSlider*>( self );
	} else {
		delete self;
	}
}
